src/main/java/sparksoniq/jsoniq/runtime/iterator/SparkRuntimeIterator.java

```diff
@@ -113,4 +110,22 @@ public Item next() {
         currentResultIndex++;
         return item;
     }
+
+    @Override
+    public JavaRDD<Item> getRDD(DynamicContext context) {
+        if (isDataFrame()) {
+            Dataset<Row> df = this.getDataFrame(context);
+            JavaRDD<Row> rowRDD = df.javaRDD();
+            return rowRDD.map(new RowToItemMapper(getMetadata()));
+        } else {
+            return getRDDAux(context);
```

Comment:
As a pointer to illustrate my comment: here, this is a duplicate implementation of what is already done in the hybrid iterator -> the SparkRuntimeIterator is no longer needed, because the hybrid iterator does that job, and even better.